packages/beacon-node/src/sync/range/utils/peerBalancer.ts

```diff
@@ -29,20 +28,26 @@ export class ChainPeersBalancer {
    * Return the most suitable peer to retry
    * Sort peers by (1) no failed request (2) less active requests, then pick first
    */
-  bestPeerToRetryBatch(batch: Batch): PeerIdStr | undefined {
+  bestPeerToRetryBatch(batch: Batch): PeerSyncMeta | undefined {
     if (batch.state.status !== BatchStatus.AwaitingDownload) {
       return;
     }
     const {partialDownload} = batch.state;
 
     const failedPeers = new Set(batch.getFailedPeers());
     const sortedBestPeers = sortBy(
-      this.peers.filter((peerId) => {
+      this.peers.filter(({earliestAvailableSlot, custodyGroups}) => {
+        const earliestSlot = earliestAvailableSlot ?? 0;
```

Comment:
this means we optimistically pick a peer without earliest_available_slot
if that's the case we need to deprioritize these peers, ie if we have peers with `earliest_available_slot` with other conditions we should pick it

that could be a TODO for #7746, we refactor the logic in that PR anyway

Reply:
ooooo!!! really good catch!!  I added a TODO to the code at that location.